projects/crc-timing/testing/test.py

```
@@ -0,0 +1,138 @@
+    """
+
+    try:
+        test_artifact_dir_p = [None]
+        test_artifact_dir_p[0] = env.ARTIFACT_DIR
+
+        failed = test_crc_timing.test()
+
+        logging.info("test_crc_timing.test " + ("failed" if failed else "passed"))
+
+        if config.project.get_config("matbench.enabled"):
+            exc = generate_visualization(env.ARTIFACT_DIR)
```

Ruff (0.13.1)

Undefined name `generate_visualization`

(F821)